Scripts/Core/EditorVR.Rays.cs

```diff
@@ -441,22 +461,52 @@ internal void PreProcessRaycastSource(Transform rayOrigin)
 
 			static Node? RequestNodeFromRayOrigin(Transform rayOrigin)
 			{
-				Node? node = null;
-				var evrDeviceData = evr.m_DeviceData;
-				for (var i = 0; i < evrDeviceData.Count; i++)
+				foreach (var deviceData in evr.m_DeviceData)
 				{
-					var deviceData = evrDeviceData[i];
 					if (!deviceData.proxy.active)
 						continue;
 
 					if (deviceData.rayOrigin == rayOrigin)
+						return deviceData.node;
+				}
+
+				foreach (var kvp in evr.GetNestedModule<MiniWorlds>().rays)
```

Comment:
Cool. @dunity maybe you were thinking of the common pitfall with regular for loops where you can incur extra cost accessing the length of the collection?